app/uk/gov/hmrc/gform/models/Summary.scala

```diff
@@ -103,9 +103,10 @@ object SummaryForRender {
               List(uk.gov.hmrc.gform.views.html.snippets.summary.end_section(formTemplate._id, formId, section.title, index))
         }
       }
-      val repeatingSections = formTemplate.sections.flatMap(_.fields).map(fv => (fv.id, fv.`type`)).collect {
-        case (fieldId, group: Group) => Await.result(repeatService.getAllRepeatingGroups.map(_.getEntry[List[List[FieldValue]]](fieldId.value).getOrElse(Nil)), 10 seconds)
-      }
+      val cacheMap: Future[CacheMap] = repeatService.getAllRepeatingGroups
+      val repeatingSections: Future[List[List[List[FieldValue]]]] = Future.sequence(sections.flatMap(_.fields).map(fv => (fv.id, fv.`type`)).collect {
```

Comment:
We also have the concept of repeating sections, I think it is better to rename this val to repeatingGroups to avoid confusion